internal/cli/build/build.go

```diff
@@ -0,0 +1,75 @@
+package build
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"path/filepath"
+	"strings"
+
+	"github.com/google/go-containerregistry/pkg/authn"
+	"github.com/google/go-containerregistry/pkg/crane"
+	v1 "github.com/google/go-containerregistry/pkg/v1"
+	"github.com/google/go-containerregistry/pkg/v1/empty"
+	"github.com/google/go-containerregistry/pkg/v1/mutate"
+)
+
+type Package struct {
+	Image v1.Image
+}
+
+// Push the package to the given destination.
+//
+// This searches for docker and podman credential files and uses them.
+func (p Package) Push(ctx context.Context, destination string) error {
+	opts := []crane.Option{crane.WithContext(ctx), crane.WithAuthFromKeychain(authn.DefaultKeychain)}
+
+	if err := crane.Push(p.Image, destination, opts...); err != nil {
+		return fmt.Errorf("push: %w", err)
+	}
+
+	return nil
+}
+
+// Build the package described by sourcePath.
+func Build(sourcePath string) (*Package, error) {
+	rawLayer := map[string][]byte{}
```

Comment:
Important: please put all files into a /package directory.
I've tried it top-level at first as well, but all the linux directories like /etc /dev and sysfs are cluttering root and make loading difficult.